atkinson/errors/reporters/__init__.py

```diff
@@ -0,0 +1,33 @@
+#! /usr/bin/env python
+"""
+General error report classes
+"""
+
+from abc import ABC, abstractmethod
+
+
+class BaseReport(ABC):
+    """ Abstract base class for error reports """
+
+    @classmethod
+    @abstractmethod
+    def new(cls, title, description, config):
+        """ Create a new report and return a report object """
+
+    @classmethod
+    @abstractmethod
+    def get(cls, report_id, config):
+        """ Get a current report object by ID """
+
+    @property
+    @abstractmethod
+    def report_id(self):
+        """ The id of the report """
+
+    @abstractmethod
+    def update(self, **kwargs):
+        """ Update the data on an active report """
+
+    @abstractmethod
+    def close(self):
```

Comment:
I am not clear what it means to 'close out a report'.  This does not sound at first read to be consistent with treating a report as an object.  Is it meant to be like a file close type operation, or something else?

Reply:
On reading your implemented class, I think I get it now.  This is mostly to provide a marker in the log file to indicate current batch of related logging is complete, yes?  One thought I had though - if other things are going on while the 'report' is open, wouldn't you get potentially unexpected logs interleaved into the area you are trying to make a specific report on?

Reply:
Yes the close is to indicate that the reported issue has been resolved.

Interleaved messages are going to happen, that is the purpose of the defined formatting of `f"{title}: {data}'` for all of the emitted lines. This should help someone see the report data over time in log messages.